src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -138,7 +138,11 @@ public async Task AcceptClientAsync()
         /// <returns>True if Client is connected, false otherwise</returns>
         public bool WaitForClientConnection(int clientConnectionTimeout)
         {
-            return this.clientConnectedEvent.WaitOne(clientConnectionTimeout);
+            Stopwatch stopWatch = Stopwatch.StartNew();
```

Comment:
[cosmetic] You could use `var`.

```suggestion
            var stopWatch = Stopwatch.StartNew();
```